Docs - missed out date time functions (#20107)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/date-time-functions.md

**Parameters**

> -   `unit` — The type of interval to add. [String](../../sql-reference/data-types/string.md).

Reviewer: ```suggestion
-   `unit` — The type of interval to truncate the result. [String](../../sql-reference/data-types/string.md).
```

## date\_add {#date_add}

Adds specified date/time interval to the provided date.

**Syntax** 

``` sql
date_add(unit, value, date)
```

Aliases: `dateAdd`, `DATE_ADD`. 

**Parameters**

-   `unit` — The type of interval to add. [String](../../sql-reference/data-types/string.md).

        Supported values: second, minute, hour, day, week, month, quarter, year.
-   `value` - Value in specified unit - [Int](../../sql-reference/data-types/int-uint.md)    
-   `date` — [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).

**Returned value**

Returns Date or DateTime with `value` expressed in `unit` added to `date`. 

**Example**

```sql
select date_add(YEAR, 3, toDate('2018-01-01'));
```

```text
┌─plus(toDate('2018-01-01'), toIntervalYear(3))─┐
│                                    2021-01-01 │
└───────────────────────────────────────────────┘
```

## date\_diff {#date_diff}

Returns the difference between two Date or DateTime values.

**Syntax**

``` sql
date_diff('unit', startdate, enddate, [timezone])
```

Aliases: `dateDiff`, `DATE_DIFF`. 

**Parameters**

> -   `unit` — The type of interval to add. [String](../../sql-reference/data-types/string.md).

Reviewer: ```suggestion
-   `unit` — The type of interval for result [String](../../sql-reference/data-types/string.md).
```

        Supported values: second, minute, hour, day, week, month, quarter, year.

> -   `startdate` — The first time value to compare. [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
> 
> -   `enddate` — The second time value to compare. [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).

Reviewer: ```suggestion
-   `startdate` — The first time value to subtract (the subtrahend). [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).

-   `enddate` — The second time value to subtract from (the minuend). [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```

-   `timezone` — Optional parameter. If specified, it is applied to both `startdate` and `enddate`. If not specified, timezones of `startdate` and `enddate` are used. If they are not the same, the result is unspecified.

**Returned value**

> Difference between `startdate` and `enddate` expressed in `unit`.

Reviewer: ```suggestion
Difference between `enddate` and `startdate` expressed in `unit`.
```

-   [toStartOfInterval](#tostartofintervaltime-or-data-interval-x-unit-time-zone)

> ## date\_add {#date_add}

Reviewer 2: But how do these functions work?
What will happen if we will add one month to 31th January?

Author: I think these functions were written to make them compatible with MySQL functions. For example https://www.w3schools.com/sql/func_mysql_date_add.asp. 
So I assume that they are expected to work how they work with MySQL ?
To answer your question adding 1 month to Jan 31st:
```
select DATE_ADD(MONTH, 1, toDateTime('2021-01-31'));

┌─plus(toDateTime('2021-01-31'), toIntervalMonth(1))─┐
│                                2021-02-28 00:00:00 │
└────────────────────────────────────────────────────┘
```

Reviewer 2: https://www.w3schools.com/sql/func_mysql_date_add.asp
The type of interval to add. Can be one of the following      values:            MICROSECOND  

probably we can use the same sentence `The type of interval `

Author: Sure, reworded the def of `unit` to use `The type of interval ....` wherever relevant.

Adds specified date/time interval to the provided date.

**Syntax** 

``` sql
date_add(unit, value, date)
```

Aliases: `dateAdd`, `DATE_ADD`. 

**Parameters**

> -   `unit` - The unit of time - [String](../syntax.md#syntax-string-literal).

Reviewer 2: only it's not String

'second' -- String
second -- Literal ?

Author: yes, it's a `string literal`. Made the change.

Author: I think it's the other way around:
'second' - string literal: https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/syntax.md#string-syntax-string-literal. Example: `HOUR` `MINUTE` etc.
second - string: https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/data-types/string.md#string-string. In this case `HOUR`, `MINUTE` etc.

For all these examples, it should actually be String.

    Possible values:

    - `second`
    - `minute`
    - `hour`
    - `day`
    - `week`
    - `month`
    - `quarter`
    - `year`
-   `value` - Amount of the specified unit of time.    

> -   `date` — Date or Date with time - [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).

Reviewer: ```suggestion
-   `date` — [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```

**Example**

```sql
select date_add(YEAR, 3, toDate('2018-01-01'));
```

```text
┌─plus(toDate('2018-01-01'), toIntervalYear(3))─┐
│                                    2021-01-01 │
└───────────────────────────────────────────────┘
```

> ## date\_diff {#date_diff}

Reviewer: `dateDiff` already documented in this file https://github.com/ClickHouse/ClickHouse/blob/f7dbdc623cc4122f7b03f621e2c3f4c4d745b74f/docs/en/sql-reference/functions/date-time-functions.md#L553

You can add some information, e.g. aliases

Author: Right, good catch. Preserved the older one and added some additional info.
